packages/ui/src/ui/pages/query-tracker/QueryTrackerTopRow/index.tsx

```diff
@@ -39,15 +39,23 @@ const QueryTrackerTopRow: FC = () => {
     const handleChangeEngine = useCallback(
         (newEngine: QueryEngine) => {
             const newSettings = {...settings};
-
-            if (newEngine === QueryEngine.CHYT && settings && 'cluster' in settings) {
-                dispatch(loadCliqueByCluster(settings.cluster as string));
+            const isSpyt = newEngine === QueryEngine.SPYT;
+            const isChyt = newEngine === QueryEngine.CHYT;
+
+            if ((isSpyt || isChyt) && settings && 'cluster' in settings) {
+                dispatch(
+                    loadCliqueByCluster(
+                        newEngine === QueryEngine.SPYT ? QueryEngine.SPYT : QueryEngine.CHYT,
+                        settings.cluster as string,
+                    ),
```

Comment:
it might be a little simplified
```ts
if ((isSpyt || isChyt) && settings?.cluster) {
    dispatch(loadCliqueByCluster(newEngine, settings.cluster))
}
```